src/org/loklak/data/DAO.java

```diff
@@ -933,10 +933,19 @@ public static int deleteOld(IndexName indexName, Date createDateLimit) {
          * @param resultCount - the number of messages in the result; can be zero if only aggregations are wanted
          * @param aggregationLimit - the maximum count of facet entities, not search results
          * @param aggregationFields - names of the aggregation fields. If no aggregation is wanted, pass no (zero) field(s)
+         * @param filterList - list of filters in String datatype
          */
-        public SearchLocalMessages(final String q, final Timeline.Order order_field, final int timezoneOffset, final int resultCount, final int aggregationLimit, final String... aggregationFields) {
+        public SearchLocalMessages (
+                final String q,
+                final Timeline.Order order_field,
+                final int timezoneOffset,
+                final int resultCount,
+                final int aggregationLimit,
+                final ArrayList<String> filterList,
```

Comment:
Wouldn't it be better to use more generic `List` instead of `ArrayList`? We may wish to use something other than `ArrayList` in future.